File: products/Introduction_to_Computational_Science_Modules/02_System_Dynamics/R/R_Computational_Toolbox_Files/Diffusion_R/applyHotCold.R
applyHotCold = function(bar, hotSites, coldSites) {
# APPLYHOTCOLD return bar with hot and cold sites

utils::globalVariables(c("COLD", "HOT"))

newbar = bar

for(k in 1:length(hotSites[,1])){
	newbar[hotSites[k,1],hotSites[k,2]]<-HOT
	}
for(k in 1:length(coldSites[,1])) {
	newbar[coldSites[k,1],coldSites[k,2]]<-COLD
	}
return(newbar)
}

File: products/Introduction_to_Computational_Science_Modules/02_System_Dynamics/MATLAB/Files/Diffusion/applyHotCold.m
function bar = applyHotCold(bar, hotSites, coldSites)
% APPLYHOTCOLD return bar with hot and cold sites
global HOT COLD 

for k = 1:size(hotSites, 1)
    bar(hotSites(k, 1), hotSites(k, 2)) = HOT;
end;

for k = 1:size(coldSites, 1)
    bar(coldSites(k, 1), coldSites(k, 2)) = COLD;
end;